README.md

```diff
@@ -1,3 +1,71 @@
-# kafka-lag-monitoring-system
+# 🚀 Kafka Lag Monitoring System (Prometheus & Grafana)
 
-check pull request branch dev
\ No newline at end of file
+This project provides a complete, containerized stack for monitoring Kafka consumer lag in real-time using **Prometheus** for data collection and **Grafana** for visualization.
+
+## 📦 Project Stack
+
+| Component | Port | Description |
+| :--- | :--- | :--- |
+| **Kafka** | `9092` | The messaging broker. |
+| **Kafka Exporter** | `9308` | Exposes Kafka metrics (including lag) for Prometheus. |
+| **Prometheus** | `9090` | Time-series database that scrapes metrics from the Exporter. |
+| **Grafana** | `3000` | Dashboard tool for visualizing the lag data. |
+| **Zookeeper** | `2181` | Coordinates the Kafka cluster. |
+
+## ⚙️ Prerequisites
+
+* **Docker** and **Docker Compose** installed on your system.
+* **Python 3** and `kafka-python` installed for the test scripts: `pip install kafka-python`.
+
+## 🛠️ Setup and Installation
+
+### 1. Project Structure
+
+Ensure your project directory is set up correctly:
+
+kafka-lag-monitoring-system/ 
```

Comment:
Since the preview format looks a bit incorrect

![CleanShot 2025-11-07 at 09 35 41@2x](https://github.com/user-attachments/assets/3e4c82f0-02cf-4a0a-bd81-998acafcb7ec)

Maybe you need to cover with this block as well

```text
kafka-lag-monitoring-system/ 
    ├── docker-compose.yml 
    ├── src
        ├── kafka_consumer.py 
        ├── kafka_producer.py 
    └── prometheus/ 
        └── prometheus.yml
```